uvtools/plot.py

```diff
@@ -466,3 +469,265 @@ def omni_view_gif(filenames, name='omni_movie.gif'):
         images.append(imageio.imread(filename))
     imageio.mimsave(name, images)
     
+def plot_diff_waterfall(uvd1, uvd2, antpairpol, plot_type="all", 
+                        skip_check=False, save_path=None):
+    """Produce waterfall plot(s) of differenced visibilities.
+
+    Parameters
+    ----------
+    uvd1, uvd2 : pyuvdata.UVData
+        UVData objects which store visibilities to be differenced and their
+        associated metadata. They should have the same number of frequencies,
+        same baselines, and same times as each other.
+
+    antpairpol : tuple
+        Tuple specifying which baseline and polarization to use to compare 
+        visibility waterfalls. See pyuvdata.UVData.get_data method docstring 
+        for information on accepted tuples.
+    
+    plot_type : str, tuple of str, or list of str, optional
+        Which spaces to use for investigating differences. Available options 
+        are as follows: time and frequency ('time_vs_freq'); time and delay 
+        ('time_vs_dly'); fringe rate and frequency ('fr_vs_freq'); fringe 
+        rate and delay ('fr_vs_dly'). Default is to use all plot types.
+    
+    skip_check : bool, optional
+        Whether to check that the metadata in `uvd1` and `uvd2` match.
+        Default behavior is to check the metadata.
+
+    save_path : str, optional
```

Comment:
It's probably better to return the `matplotlib.Figure` object and allow the user to save it. It's only one extra line of code for the user, and gives them a lot more control over the figure.